larpandoracontent/LArHelpers/LArHitWidthHelper.h

```diff
@@ -0,0 +1,330 @@
+        
+        /**
+         *  @brief  Constructor
+         */
+        ClusterParameters(const pandora::Cluster *const pCluster, const unsigned int numCaloHits, const float totalWeight, const ConstituentHitVector &constituentHitVector, const pandora::CartesianVector &lowerXExtrema, const pandora::CartesianVector &higherXExtrema);
+
+        /**
+         *  @brief  Returns the address of the cluster
+         */
+        const pandora::Cluster* GetClusterAddress() const;
+
+        /**
+         *  @brief  Returns the number of calo hits within the cluster
+         */
+        unsigned int GetNumCaloHits() const;
+        
+        /**
+         *  @brief  Returns the total weight of the constituent hits
+         */
+        float GetTotalWeight() const;
+
+        /**
+         *  @brief  Returns the vector of constituent hits
+         */
+        const ConstituentHitVector& GetConstituentHitVector() const;
+
+        /**
+         *  @brief  Returns the lower x extremal point of the constituent hits
+         */
+        const pandora::CartesianVector& GetLowerXExtrema() const;
+
+        /**
+         *  @brief  Returns the higher x extremal point of the constituent hits
+         */
+        const pandora::CartesianVector& GetHigherXExtrema() const;     
+
+    private:
+        const pandora::Cluster *m_pCluster;                  ///< The address of the cluster 
+        const unsigned int m_numCaloHits;                    ///< The number of calo hits within the cluster
+        const ConstituentHitVector m_constituentHitVector;   ///< The vector of constituent hits
+        const float m_totalWeight;                           ///< The total hit weight of the contituent hits
+        const pandora::CartesianVector m_lowerXExtrema;      ///< The lower x extremal point of the constituent hits
+        const pandora::CartesianVector m_higherXExtrema;     ///< The higher x extremal point of the constituent hits
+    };
+
+    typedef std::map<const pandora::Cluster*, const ClusterParameters> ClusterToParametersMap;
+
+    /**
+     *  @brief  ClusterToParametersMapStore class
+     */
+    class ClusterToParametersMapStore
+    {
+    public:
+        /**
+         *  @brief  Delete the default copy constructor
+         */
+        ClusterToParametersMapStore(ClusterToParametersMapStore const& obj) = delete;
+
+        /**
+         *  @brief  Delete the default move constructor
+         */
+        ClusterToParametersMapStore(ClusterToParametersMapStore&& obj) = delete;
+        
+        /**
+         *  @brief  Returns a pointer to the ClusterToParametersMapStore instance 
+         */
+        static ClusterToParametersMapStore* Instance();
+
+        /**
+         *  @brief  Returns the map [cluster -> cluster parameters]
+         */
+        ClusterToParametersMap& GetMap();
+
+    private:
+        /**
+         *  @brief  Default constructor
```

Comment:
No particular harm in it, but in general you only add doxygen style comments to your public/protected interface, rather than private functions.